packages/mui-material/src/useMediaQuery/useMediaQuery.ts

```diff
@@ -94,15 +90,19 @@ function useMediaQueryNew(
   defaultMatches: boolean,
   matchMedia: typeof window.matchMedia | null,
   ssrMatchMedia: ((query: string) => { matches: boolean }) | null,
+  noSsr: boolean | undefined,
 ): boolean {
   const getDefaultSnapshot = React.useCallback(() => defaultMatches, [defaultMatches]);
   const getServerSnapshot = React.useMemo(() => {
+    if (noSsr && matchMedia) {
+      return () => matchMedia!(query).matches;
+    }
```

Comment:
Fix a regression, between the "old" and "new" version (depending on the version of React used). To be fair, this can be a different PR. It's not needed to fix the bug in the blog.